src/averageAges.js

```diff
@@ -55,8 +71,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(
+      (person) =>
+        people.find((mother) => person.mother === mother.name)
+          && person.sex === 'm'
+    )
+    : people.filter((person) =>
+      people.find((mother) => person.mother === mother.name)
+    );
+
+  const different = children.map(
+    (child) =>
+      child.born - people.find((mother) => child.mother === mother.name).born
```

Comment:
apply de-structuring of arguments for practice sake
`(child)` -> `({ born, mother })`
`(mother)` -> `({ name })`